### ISSUE TITLE
Non GPU version

### ISSUE THREAD

**Issue by bhmchai** [rpautrat/SuperPoint#41]
Is it possible to run without GPU. How to run only on CPU?

**Comment by rpautrat** [comment]
Hi, it is indeed not possible to train on CPU for now. But the training would be quite long with only a CPU...
However once it has been trained, it is possible to export the weights and the graph to a file, so that you can reload it later (with CPU). I wanted to publish those weights, but it seems that something is wrong with the export and the exported network does not really work. I am still investigating this.

**Comment by shylockyuan** [comment]
Hello,  I trained the SuperPoint with 1080,8G.It turns out that graphics card memory is not enough. So could you tell me what size of graphics card memory do I need? Thank you!

**Comment by rpautrat** [comment]
Hi, I personally managed to train also with a 1080 but with 10 Gb. And I had a batch size of 3, so if you use a batch size of 2, your graphic card should already be enough normally.

**Comment by shylockyuan** [comment]
Hi!
I tried your advice but it didn’t work. So I replace 1080(8G) with 1080ti(11G) and it failed(both try batch size 2 and batch size 3). I don’t know how to solve it. Could you give me some advice?Thanks a lot.
experiment.py:142: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
[04/16/2019 20:48:06 INFO] Running command TRAIN
[04/16/2019 20:48:06 INFO] Number of GPUs detected: 1
[04/16/2019 20:48:12 INFO] Caching data, fist access will take some time.
[04/16/2019 20:48:13 INFO] Caching data, fist access will take some time.
[04/16/2019 20:48:14 INFO] Caching data, fist access will take some time.
2019-04-16 20:48:15.534189: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-04-16 20:48:15.764607: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-04-16 20:48:15.764916: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1212] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.58GiB
2019-04-16 20:48:15.764930: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1312] Adding visible gpu devices: 0
2019-04-16 20:48:19.519269: I tensorflow/core/common_runtime/gpu/gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10239 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
[04/16/2019 20:48:19 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:20 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
[04/16/2019 20:48:21 INFO] Scale of 0 disables regularizer.
2019-04-16 20:48:21.731789: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1312] Adding visible gpu devices: 0
2019-04-16 20:48:21.731968: I tensorflow/core/common_runtime/gpu/gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 321 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
[04/16/2019 20:48:31 INFO] Start training
2019-04-16 20:48:38.222588: I tensorflow/core/kernels/cuda_solvers.cc:159] Creating CudaSolver handles for stream 0x7ffb02e7f7b0
[04/16/2019 20:49:30 INFO] Iter    0: loss 28.2286, precision 0.0026, recall 0.0080
[04/16/2019 20:53:18 INFO] Iter 1000: loss 8.2341, precision 0.0073, recall 0.0240
[04/16/2019 20:57:12 INFO] Iter 2000: loss 8.6386, precision 0.0094, recall 0.0304
[04/16/2019 21:01:13 INFO] Iter 3000: loss 7.1039, precision 0.0105, recall 0.0340
Killed

**Comment by rpautrat** [comment]
Your problem is actually not the GPU, your program has been rather killed for a lack of temporary memory I think. I guess that you have the parameter  'cache_in_memory' set to True in your config file `superpoint_coco.yaml`? Try to set it to False to solve the problem (or pull the last version of the master branch, I did the correction a few days ago).
